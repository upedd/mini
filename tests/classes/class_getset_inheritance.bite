native print;

class Foo {
    get x {
        print("foo.x");
        return 10;
    }

    set x(value) {
        print("=foo.x");
    }
}

class Bar : Foo {
    override get x {
        print("bar.x");
        return 20;
    }

    override set x(value) {
        print("=bar.x");
    }

    get_super_x() {
        return super.x;
    }

    #set_super_x() {
    #    super.x = 10;
    #}
}

let instance = Bar();

instance.x = 10;
print(instance.x);

#instance.set_super_x();
print(instance.get_super_x());

instance.x = 10;
print(instance.x);