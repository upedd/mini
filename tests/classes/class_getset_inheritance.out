=bar.x
bar.x
20
=foo.x
foo.x
10
=bar.x
bar.x
20